File: patterns/c-patterns/pattern79.c
// Pattern 79: ...

// *
//  *
//   *
//    *
//   *
//  *
// *

#include <stdio.h>

int main()
{
    int n = 7;
    int i, j;

    int x = 1;
    for (i = 1; i <= n; i++)
    {
        for (j = 1; j <= x; j++)
        {
            if (j == x)
                printf("* ");
            else
                printf(" ");
        }
        printf("\n");
        if (i <= n / 2)
            x++;
        else
            x--;
    }
    return 0;
}

File: patterns/python-patterns/pattern79.py
# Pattern 79: ...

# *
#  *
#   *
#    *
#   *
#  *
# *

n = 7
x = 1
for i in range(1, n + 1):
    for j in range(1, x + 1):
        if j == x:
            print("* ", end="")
        else:
            print(" ", end="")
    print()
    if i <= n / 2:
        x += 1
    else:
        x -= 1
